src/cdk/tree/tree.spec.ts

```diff
@@ -0,0 +1,201 @@
+
+import {CollectionViewer, DataSource} from '@angular/cdk/collections';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {Observable} from 'rxjs/Observable';
+import {combineLatest} from 'rxjs/observable/combineLatest';
+import {map} from 'rxjs/operator/map';
+
+import {TreeControl} from './control/tree-control';
+import {FlatTreeControl} from './control/flat-tree-control';
+import {CdkTreeModule} from './index';
+import {CdkTree} from './tree';
+
+
+describe('CdkTree', () => {
+  let fixture: ComponentFixture<SimpleCdkTreeApp>;
+
+  let component: SimpleCdkTreeApp;
+  let dataSource: FakeDataSource;
+  let tree: CdkTree<any>;
+  let treeElement: HTMLElement;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [CdkTreeModule],
+      declarations: [
+        SimpleCdkTreeApp,
+        // TODO(tinayuangao): Add more test cases with the cdk-tree
+        //   DynamicDataSourceCdkTreeApp,
+        //   NodeContextCdkTreeApp,
+        //   WhenNodeCdkTreeApp
+      ],
+    }).compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(SimpleCdkTreeApp);
+
+    component = fixture.componentInstance;
+    dataSource = component.dataSource as FakeDataSource;
+    tree = component.tree;
+    treeElement = fixture.nativeElement.querySelector('cdk-tree');
+
+    fixture.detectChanges();
+  });
+
+  describe('should initialize', () => {
+    it('with a connected data source', () => {
+      expect(tree.dataSource).toBe(dataSource);
+      expect(dataSource.isConnected).toBe(true);
+    });
+
+    it('with rendered nodes', () => {
+      const nodes = getNodes(treeElement);
+
+      expect(nodes).not.toBe(undefined);
+      expect(nodes[0].classList).toContain('customNodeClass');
+    });
+
+    it('with the right accessibility roles', () => {
+      expect(treeElement.getAttribute('role')).toBe('tree');
+
+      getNodes(treeElement).forEach(node => {
+        expect(node.getAttribute('role')).toBe('treeitem');
+      });
+    });
+
+    it('with the right data', () => {
+      expect(dataSource.data.length).toBe(3);
+
+      let data = dataSource.data;
+      expectFlatTreeToMatchContent(treeElement,
+          [
+            `${data[0].a} - ${data[0].b} + ${data[0].c}`,
+            `${data[1].a} - ${data[1].b} + ${data[1].c}`,
+            `${data[2].a} - ${data[2].b} + ${data[2].c}`
+          ],
+          [1, 1, 1]);
+
+      dataSource.addData(2);
+      fixture.detectChanges();
+
+      data = dataSource.data;
+      expect(data.length).toBe(4);
+      expectFlatTreeToMatchContent(treeElement,
+        [
+          `${data[0].a} - ${data[0].b} + ${data[0].c}`,
+          `${data[1].a} - ${data[1].b} + ${data[1].c}`,
+          `${data[2].a} - ${data[2].b} + ${data[2].c}`,
+          `${data[3].a} - ${data[3].b} + ${data[3].c}`
+        ],
+        [1, 1, 1, 2]);
+    });
+  });
+});
+
+export class TestData {
+  a: string;
```

Comment:
Make up fake properties for this? I'm a fan of _obviously fake_ testing data like `ponyName` or `pizzaTopping` since it  is clearly distinguished from actual component APIs